lib/streamlit/elements/skeleton.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Optional, cast
+
+from streamlit.proto.Skeleton_pb2 import Skeleton as SkeletonProto
+from streamlit.runtime.metrics_util import gather_metrics
+
+if TYPE_CHECKING:
+    from streamlit.delta_generator import DeltaGenerator
+
+
+class SkeletonMixin:
+    @gather_metrics("skeleton")
+    def skeleton(
+        self, height: Optional[int] = None, *, width: Optional[float] = None
```

Comment:
nit: we have updated our codebase to the new/future types of annotations yesterday. To use that would need:

Adding: `from __future__ import annotations` above the `from typing...` and replacing the `Optional` use with `int | None`.

Reply:
Also, we have the policy to add almost all new parameters as `keyword-only` in the beginning and eventually move them to positional usage later. I think in this case, it could be discussed if `height` is prominent enough to directly provide as positional. But it is a safer choice to move it to keyword-only parameter for now.